x.py

```diff
@@ -92,6 +93,20 @@ def check_version(current: str, required: Tuple[int, int, int], prog_name: Optio
 
     return semver
 
+def prepare() -> None:
+    basedir = Path(__file__).parent.absolute()
+    # install git hooks
+    git_hooks_dir = basedir / ".git/hooks/"
+    kvrocks_git_hooks_dir = basedir / "utils/git-hooks/"
+    os.makedirs(git_hooks_dir, exist_ok=True)
+    for hook in os.listdir(kvrocks_git_hooks_dir):
+        dst = os.path.join(git_hooks_dir, hook.split('.')[0])
+        hook_path = os.path.join(kvrocks_git_hooks_dir, hook)
+        if os.path.exists(dst):
+            os.remove(dst)
+        os.symlink(hook_path, dst)
+        os.chmod(dst, os.stat(dst).st_mode | stat.S_IEXEC)
+        print(hook.split('.')[0], "installed at", os.path.abspath(dst))
```

Comment:
This patch is good to merge, while:

1. We can keep use `pathlib` instead of sometimes `os.path.join`.
2. We can chmod for the shell script checked in directly.
3. We can name the shell script as `pre-push` directly.